include/rocksdb/utilities/object_registry.h

```diff
@@ -47,54 +46,148 @@ class ObjectLibrary {
   class Entry {
    public:
     virtual ~Entry() {}
-    Entry(const std::string& name) : name_(std::move(name)) {}
+    virtual bool Matches(const std::string& target) const = 0;
+    virtual const char* Name() const = 0;
+  };
+
+  // Class for matching target strings to a pattern.
+  // Entries consist of a name that starts the pattern and attributes
+  // The following attributes can be added to the entry:
+  //   -Suffix: Comparable to name(pattern)
+  //   -Pattern: Comparable to name(pattern).+
```

Comment:
I also find it confusing that "pattern" is used in three ways: PatternEntry (general), Pattern subtype, and pattern string to match. The last I think we can call "separator" as in my discussion about difference vs. regexes.

Reply:
Renamed "AddPattern" and associated variables to "Separator".  The class is still called PatternEntry but most other references have been removed.